src/components/molecules/PollBox/PollBox.tsx

```diff
@@ -21,23 +21,20 @@ const DEFAULT_VALUES = {
   questions: [DEFAULT_QUESTION, DEFAULT_QUESTION],
 };
 
-export const PollBox: React.FC<PollBoxProps> = ({ createPoll }) => {
-  const { register, control, handleSubmit, reset, watch } = useForm<PollValues>(
-    {
-      defaultValues: DEFAULT_VALUES,
-    }
-  );
+export const PollBox: React.FC<PollBoxProps> = ({ onPollSubmit }) => {
+  const { register, control, handleSubmit, watch } = useForm<PollValues>({
+    defaultValues: DEFAULT_VALUES,
+  });
   const { fields, append } = useFieldArray({ name: "questions", control });
   const [question1, question2] = watch("questions");
   const topic = watch("topic");
 
-  const onPollSubmit = handleSubmit(({ topic, questions }) => {
-    createPoll({
+  const handlePollSubmit = handleSubmit(({ topic, questions }) =>
+    onPollSubmit({
       topic,
       questions: questions.map(({ name }, id) => ({ name, id })),
-    });
-    reset();
```

Comment:
[NB] Is this `reset()` definitely no longer needed here?

Reply:
It looks like this whole `PollBox` component will be unrendered when `activeOption` is no longer `ChatOptionType.poll`, so I expect that will 'clear the old data' appropriately. 👌🏻